gazelle/python/testdata/per_file_non_empty_init/foo.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# For test purposes only.
```

Comment:
I would love to see a test how it behaves if there are multiple directories where the layout could be:
```
//:foo.py # contains from bar import my_func
//bar/__init__.py # non-empty, from bar.foo import my_func
//bar/foo.py # non-empty, has def my_func
//baz/__init__.py
//baz/foo.py # imports from bar.foo import my_func
```

Reply:
ok that revealed 2 bugs:
1. `__init__.py`s in subdirs weren't being stat-ed at all because I wasn't prepending `args.Dir`
2. if we have `baz/baz.py`, we generate `name = "baz", src = ["baz/baz.py"]`, which seems pretty problematic. I decided to just not traverse subdirs in per-file generation mode